mailpoet/lib/Tasks/Subscribers/BatchIterator.php

```diff
@@ -24,6 +28,7 @@ public function __construct(
     }
     $this->taskId = (int)$taskId;
     $this->batchSize = (int)$batchSize;
+    $this->scheduledTaskSubscribersRepository = ContainerWrapper::getInstance()->get(ScheduledTaskSubscribersRepository::class);
```

Comment:
I'm not sure about this. For me, it looks like a hidden dependency. Did you consider passing the repository via constructor?

Reply:
Yeah, I considered passing the repository via the constructor. I opted not to as we instantiate BatchIterator directly in a few places and I didn't want to mess with that. I don't really see a problem with getting an instance of ScheduledTaskSubscribersRepository here directly and we can always change that in the future if needed. But I'd be happy to change it now if you see a reason for it.